### PR TITLE
Pandora 112 PR for issue 112

### PR DESCRIPTION
**Pull request by tamojitdas**
For issue #112 :: HI @WJ-van-Hoek  Please review the code changes made for Increasing Coverage report for LocalDateServiceImple.class .. In the process some test classes are also created ...
Let me know if there are any changes (addition/deletion) I need to further work on. 
Thanks.

### CODE REVIEW

File: pandora-calculation/src/main/java/com/science/earth/biogeochemistry/freshwaters/pandora/controllers/requestbodies/CellCalculationRequest.java [hunk 1]
```diff
@@ -38,5 +38,6 @@ public void validate() throws ValidationException {
 	}
 
 	// Additional validation or consistency checks can be added here
+        //comment
```
Comment (WJ-van-Hoek) [inline pandora-calculation/src/main/java/com/science/earth/biogeochemistry/freshwaters/pandora/controllers/requestbodies/CellCalculationRequest.java:41]: this class is out of scope for this PR

File: pandora-calculation/src/main/resources/test.properties [hunk 2]
```diff
@@ -0,0 +1,28 @@
+dic.enabled=true
```
Comment (WJ-van-Hoek) [inline pandora-calculation/src/main/resources/test.properties:1]: specific test.properties should not be required for the issue in the ticket

File: pandora-calculation/src/main/resources/test1.properties [hunk 3]
```diff
@@ -0,0 +1,28 @@
+dic.enabled=true
```
Comment (WJ-van-Hoek) [inline pandora-calculation/src/main/resources/test1.properties:1]: specific test1.properties should not be required for the issue in the ticket

**Review by WJ-van-Hoek — CHANGES_REQUESTED** [review]
Thank you a lot for the work you have been doing!

Unfortunately, I suggest to start the ticket from scratch again as too much redundant code is written for the sole purpose of the test. CellTimestepServiceImpl and properties should not be part of code changes here.

As we see, the LocalDateTimeServiceImpl class has two public methods:
calculateTEndAsLocalDateTime and getDateTimeList

I suggest to make a LocalDateTimeServiceImplTest class where these methods are unit tested. The private method (generateDateTimes) does not need to be unit tested. If you need more guidance, let me know.

**Comment by tamojitdas** [comment]
Sure, will do the changes .. I have some Idea On my mind

File: pandora-calculation/src/main/resources/test1.properties (hunk 3 above)
Comment (tamojitdas) [reply pandora-calculation/src/main/resources/test1.properties]: Deleted test1.properties

File: pandora-calculation/src/main/resources/test.properties (hunk 2 above)
Comment (tamojitdas) [reply pandora-calculation/src/main/resources/test.properties]: Deleted test.properties

File: pandora-calculation/src/main/java/com/science/earth/biogeochemistry/freshwaters/pandora/controllers/requestbodies/CellCalculationRequest.java (hunk 1 above)
Comment (tamojitdas) [reply pandora-calculation/src/main/java/com/science/earth/biogeochemistry/freshwaters/pandora/controllers/requestbodies/CellCalculationRequest.java]: Reverted

**Comment by tamojitdas** [comment]
Hi @WJ-van-Hoek  pls re-review the code changes.

**Comment by tamojitdas** [comment]
hi @WJ-van-Hoek , Pls re-Review the changes. Thanks....Was bit busy this week..Couldn't get to finish it earlier.

File: pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/CellTimestepServiceImplTest.java [hunk 4]
```diff
@@ -88,6 +88,8 @@ void setUp() throws Exception {
     @Test
     void testCalculateNextTimestep() {
 	//given
+//        System.setProperty("pandora.timestep.unit", "month");
+//        System.setProperty("pandora.timestep.length", "1L");
```
Comment (WJ-van-Hoek) [inline pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/CellTimestepServiceImplTest.java:92]: please remoe

File: pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/CellTimestepServiceImplIntegrationTest.java
Comment (WJ-van-Hoek) [inline pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/CellTimestepServiceImplIntegrationTest.java:1]: please remove all changes in this class, this class is out of scope for the ticket

File: pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/LocalDateTimeServiceTest.java
Comment (WJ-van-Hoek) [inline pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/LocalDateTimeServiceTest.java:1]: @tamojitdas
First of all, thanks a lot for your work. Due to private commitments I did not have time to work on your PR earlier.

Please find my code changes for LocalDateTimeServiceTest for PANDORA-112. If this is applied, together with my other comments in the PR it is ready to be merged. You can pull this change directly in your local PANDORA-112 branch.

File: pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/CellTimestepServiceImplTest.java (hunk 4 above)
Comment (tamojitdas) [reply pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/CellTimestepServiceImplTest.java:92]: Removed the comments

File: pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/CellTimestepServiceImplIntegrationTest.java
Comment (tamojitdas) [reply pandora-calculation/src/test/java/com/science/earth/biogeochemistry/freshwaters/pandora/general/services/implementations/CellTimestepServiceImplIntegrationTest.java:1]: Hi @WJ-van-Hoek  reverted the changes pls check

**Comment by tamojitdas** [comment]
Hi @WJ-van-Hoek  please find my changes as per your latest review.

**Review by WJ-van-Hoek — APPROVED** [review]

**Comment by WJ-van-Hoek** [comment]
> Hi @WJ-van-Hoek please find my changes as per your latest review.

@tamojitdas thank you for your time! I merged the PR
